private/bufpkg/bufprotosource/message_range.go

```diff
@@ -14,8 +14,14 @@
 
 package bufprotosource
 
+import "math"
+
 const (
-	messageRangeInclusiveMax = 536870911
+	// MessageRangeInclusiveMax is the maximum allowed tag for a message field.
+	MessageRangeInclusiveMax = 536870911 // 2^29 - 1
```

Comment:
Note to self, if we're okay making this exported, I should reference it in #2913 (instead of [redefining it](https://github.com/bufbuild/buf/pull/2913/files#diff-59d79b7b8b1bea1b73c878e3b8749e826378a94595cc236c57913f54ead46d97R27))

Reply:
Actually, we don't want `protoencoding` to import `bufprotosource`. So I'll leave this constant duplicated for now. Maybe we can consolidate them into a different package later...